src/main.c

```diff
@@ -0,0 +1,222 @@
+	}
+
+	if (bt_le_scan_stop()) {
+        printk("Scanning successfully stopped\n");
+		return;
+	}
+
+    is_connecting = true;
+	err = bt_conn_le_create(addr, BT_CONN_LE_CREATE_CONN,
+				BT_LE_CONN_PARAM_DEFAULT, &conn_connecting);
+	if (err) {
+        is_connecting = false;
+		printk("Create conn to %s failed (%d)\n", addr_str, err);
+		start_scan();
+	}
+}
+
+static void start_scan(void)
+{
+	int err;
+
+	/* This demo doesn't require active scan */
+	err = bt_le_scan_start(BT_LE_SCAN_PASSIVE, device_found);
+	if (err) {
+		printk("Scanning failed to start (err %d)\n", err);
+		return;
+	}
+
+	printk("Scanning successfully started\n");
+}
+
+static void connected(struct bt_conn *conn, uint8_t err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+    is_connecting = false;
+	if (err) {
+		printk("Failed to connect to %s (%u)\n", addr, err);
+
+		bt_conn_unref(conn_connecting);
+		conn_connecting = NULL;
+
+		start_scan();
+		return;
+	}
+
+    conn_connecting = NULL;
+
+	conn_count++;
+
+	printk("Connected: %s\n", addr);
+
+#if defined(CONFIG_BT_SMP)
+    int sec_err = bt_conn_set_security(conn, BT_SECURITY_L2);
+
+    if (sec_err) {
+        printk("Failed to set security (%d).\n", sec_err);
+    }
+#endif
+
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	printk("Disconnected: %s (reason 0x%02x)\n", addr, reason);
+
+	bt_conn_unref(conn);
+	
+	conn_count--;
+}
+
+#if defined(CONFIG_BT_SMP)
+static void security_changed(struct bt_conn *conn, bt_security_t level,
+			     enum bt_security_err err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (!err) {
+		printk("Security changed: %s level %u\n", addr, level);
+	} else {
+		printk("Security failed: %s level %u err %d\n", addr, level,
+		       err);
+	}
+}
+#endif
+
+
+static struct bt_conn_cb conn_callbacks = {
+	.connected = connected,
+	.disconnected = disconnected,
+
+#if defined(CONFIG_BT_SMP)
+	.security_changed = security_changed,
+#endif
+};
+
+int main(void)
+{
+	int err;
+
+    err = settings_subsys_init();
+    if (err) {
+		printk("Settings subsystem init failed (err %d)\n", err);
+		return 0;
+	}
+
+    err = settings_load();
+    if (err) {
+		printk("Bluetooth settings failed to load (err %d)\n", err);
+		return 0;
+	}
+
+	err = bt_enable(NULL);
+	if (err) {
+		printk("Bluetooth init failed (err %d)\n", err);
+		return 0;
+	}
+
+	printk("Bluetooth initialized\n");
+
+    bt_conn_cb_register(&conn_callbacks);
+
+	start_scan();
+
+    while (true) {
+        
+        // take breaks inbetween scanning to not overload system
+
+        k_sleep(K_MSEC(100));
+
+        // initiate a scan if max device count is not reached and a device is not currently being connected.
+        if (conn_count < CONFIG_BT_MAX_CONN && !is_connecting) {
```

Comment:
@jakejunk150 I think you should make a `work_delayable` or `work` to start new scan after you created the connection. Can you change it?

Reply:
@trungkiena6 Would you be able to elaborate more on this one? When you say make a `work_delayable` or `work`, do you mean a function or a variable? And, if so, should I get rid of the infinite while loop? Thank you!

Reply:
@jakejunk150 
The BLE processes will follow step:
- Scan -> Found -> Connecting -> Connected (The callback function `connected`) -> Internal handler -> Disconnected (The callback function `disconnected`)
There is one rule. From scan to connected event, you can't activate the multiple scan. 
So, if you want to support multiple connection, you will need to re-scan after the current device connected.
The idea to handle is:
- Use the method k_work_delayable (https://docs.zephyrproject.org/latest/kernel/services/threads/workqueue.html) to define a work. The function work will scan device by checking number of `com_count`
- In `connected` function, you will re-scan it by trigger the function `k_work_schedule`.

Reply:
@trungkiena6 I have made another commit to implement k_work_delayable and also updated find_device_name to filter by `"DMX"`. I am not super confident that I utilized k_work_delayable correctly, but I am happy to update it again! Thanks Kien